### PR TITLE
Release 0.7.0

### PR DESCRIPTION
**Pull request by ianbrown9475**

### CODE REVIEW

**Review by jusjusjus — DISMISSED** [review]

File: bin/mffdiff.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    print(f">>> {status} @ reader_instance.{prop} "
+          f"{': ' + info if info else ''}")
+    if msg:
+        print(msg)
+
+    return left_value if status == 'match' else None
+
+
+simple_props = ['flavor', 'sampling_rates', 'durations', 'startdatetime',
```
Comment (ephathaway) [inline bin/mffdiff.py:99]: Let's use 'mff_flavor' instead.

File: mffpy/xml_files.py [hunk 2]
```diff
@@ -289,6 +289,19 @@ def _parse_calibration(self, cali):
         }
         return ans
 
+    @cached_property
+    def channels(self) -> List[Dict[str, Any]]:
+        """returns list of good and bad channels"""
```
Comment (ephathaway) [inline mffpy/xml_files.py:294]: As it is written now, it seems like this function only finds bad channels listed in the data info XML. Maybe rename this method to `bad_channels`? From the changelog, it looks like this is the intended funcitonality.

**Review by ephathaway — CHANGES_REQUESTED** [review]
Just a couple minor things, then should be good to go.

File: mffpy/xml_files.py (hunk 2 above)
Comment (ianbrown9475) [reply mffpy/xml_files.py:294]: Hm, just looking at the code, it seems like it would fetch all channels, not just bad ones. Can you explain?

**Review by jusjusjus — APPROVED** [review]

File: mffpy/xml_files.py (hunk 2 above)
Comment (ephathaway) [reply mffpy/xml_files.py:294]:
```
>>> import mffpy
>>> dinfo = mffpy.XML.from_file('examples/example_1.mff/info1.xml')
>>> dinfo.channels
[{'channels': [31, 47, 65, 79], 'exclusion': 'badChannels'}]
```

The way it is written, `DataInfo.channels` will parse the text in this `channels` tag, which looks something like this

```
<channels exclusion="badChannels">31 47 65 79</channels>
```

This is what the MFF API docs from EGI say about this tag:

This is a comma separated list of channels that have some "attribute" to take into account
* exclusion
    * goodChannels
    * badChannels

That's not super descriptive, but in my experience, I have only ever seen bad channels listed in this tag. I am guessing Damian added this method to get the bad channels for showing in sourcerer? Can you comment @damian5710?

Comment (jusjusjus) [reply mffpy/xml_files.py:294]: I haven't seen the list being comma separated in my MFF examples.  The property is called `.channels` b/c the XML tag is called `<channels>...`.

Comment (ephathaway) [reply mffpy/xml_files.py:294]: I suppose the docstring is technically true, but to me it sounds like it's going to return a list of all channels. Maybe the docstring could just be 
```
"""return contents of <channels> element"""
```

Comment (ephathaway) [reply mffpy/xml_files.py:294]: > I haven't seen the list being comma separated in my MFF examples. The property is called `.channels` b/c the XML tag is called `<channels>...`.

Yeah, I don't super trust those EGI docs. They're old.

**Pull request merged**